### DISCUSSION TITLE
notification-controller: post deployment trigger

### DISCUSSION THREAD

**Discussion by medkbadri** [fluxcd/flux2#109] [Q&A] [answered]
Hello,

First, I want to thank you for your contribution to the GitOps Ecosystem.
Just want to know, if we can use this operator in order to trigger post-deployment action.

For example, once a deployment is achieved by Flux, I want another deployment to take place.

Thanks in advance.

**Comment by stefanprodan** [comment]
> For example, once a deployment is achieved by Flux, I want another deployment to take place.

Yes, this is covered by https://github.com/fluxcd/kustomize-controller/blob/master/docs/spec/v1alpha1/kustomization.md#kustomization-dependencies

**Comment by medkbadri** [comment, marked as answer]
Thank you, Stefan, for getting back to me
I started experimenting with the new gitops toolkit and I must admit that it's really cool!
I'm targeting the following use case:
- Deployment A: a new rollout update
- Workload B (Example: Tekton pipeline that will update an API GW once the Deployment A is rolled out )  (K8S manifests unchanged in Git)

I'm afraid Kustomization dependencies won't suit this use case.

Side question: is it on your radar to implement deployment sync based on image tags like the old Flux

Thanks

**Reply by stefanprodan** [reply]
> I'm targeting the following use case:
> * Deployment A: a new rollout update
> * Workload B (Example: Tekton pipeline that will update an API GW once the Deployment A is rolled out ) (K8S manifests unchanged in Git)

In this case, you could make use of the [Event API](https://toolkit.fluxcd.io/components/notification/event/). Assuming you have a `Kustomization` that contains `Deployment A`, you can define a health check for that deployment. The kustomize-controller will issue an event after the rollout update is completed. The notification-controller receives that event in JSON format and can forward it to an HTTP endpoint. You can develop an HTTP server that receives events and based on the payload, triggers the Tekton pipelines. The event payload contains the `Kustomization` object, the Source revision that triggered it `<branch>/<commit>` and the Deployment `name/namespace` that passed or failed the health check.

**Reply by medkbadri** [reply]
I proceeded with a different approach using only Kustomize controller dependencies
![image](https://user-images.githubusercontent.com/40020831/87834778-29749a80-c883-11ea-9537-fd20d3448192.png)

Here, tekton pipeline Kustomization depends on Deployment A Kustomization

But, will definitely try your approach (more elegant)

Thanks!

**Comment by stefanprodan** [comment]
> is it on your radar to implement deployment sync based on image tags like the old Flux

Yes, the image update controller is listed on our roadmap here https://toolkit.fluxcd.io/roadmap/#flux-image-update-feature-parity
